### PR TITLE
Basic solution structure.

### PR DESCRIPTION
**Pull request by gkocik**
Basic solution structure for visual studio. Sharpmake as a build syst…em. Googletest as unit test framework.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "vendor/googletest"]
```
Comment (satirev) [inline .gitmodules:1]: Not sure if we want keep dependencies as submodules, but for now it can stay as it is.

File: src/sdk/sdk.sharpmake.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			conf.IncludePaths.Add(@"[project.SharpmakeCsPath]\include");
+			conf.Defines.Add("MNEMOSYNE_SDK_API_EXPORT");
+			
+			if (target.OutputType == OutputType.Dll)
+			{
+				conf.Output = Configuration.OutputType.Dll;
+			}
+			else
+			{
+				conf.Output = Configuration.OutputType.Lib;
```
Comment (satirev) [inline src/sdk/sdk.sharpmake.cs]: Afaik this is not needed as you set OutputType to be DLL only

File: src/sdkUnitTests/sdkUnitTests.sharpmake.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
+using Sharpmake;
+
+[module: Include("../sdk/sdk.sharpmake.cs")]
+
+namespace MnemosyneSdk
+{
+	[Generate]
+	public class SdkUnitTests : Project
```
Comment (satirev) [inline src/sdkUnitTests/sdkUnitTests.sharpmake.cs]: What about introducing base class for both lib and PE and set common properties there?

File: src/sdkUnitTests/src/main.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+TEST(UnitTestTests, ExpectNotEqual)
+{
+	EXPECT_NE(Foo(), 0);
+	EXPECT_NE(Foo(), 1);
+	EXPECT_NE(Foo(), 6);
+}
+
+
+int main(int argc, char** argv) {
```
Comment (satirev) [inline src/sdkUnitTests/src/main.cpp]: Put curly bracket in the new line XD

File: src/sdkUnitTests/src/main.cpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2020 Mnemosyne
+
+#include "function.h"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "src/gtest-all.cc"
+#include "src/gmock-all.cc"
```
Comment (satirev) [inline src/sdkUnitTests/src/main.cpp]: Include what use use ;> gmock is not required at the moment

File: vendor/Sharpmake/Sharpmake.Application.exe.config [hunk 6]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+    <runtime>
+        <gcServer enabled="true"/>
```
Comment (satirev) [inline vendor/Sharpmake/Sharpmake.Application.exe.config]: What does it do?

**Review by satirev — CHANGES_REQUESTED** [review]

File: vendor/Sharpmake/Sharpmake.Application.exe.config (hunk 6 above)
Comment (gkocik) [reply vendor/Sharpmake/Sharpmake.Application.exe.config]: Generated when compiling Sharpmake. Google says: Specifies whether the common language runtime runs server garbage collection.

File: src/sdkUnitTests/src/main.cpp (hunk 5 above)
Comment (gkocik) [reply src/sdkUnitTests/src/main.cpp]: done

File: src/sdkUnitTests/src/main.cpp (hunk 4 above)
Comment (gkocik) [reply src/sdkUnitTests/src/main.cpp]: done

File: src/sdkUnitTests/sdkUnitTests.sharpmake.cs (hunk 3 above)
Comment (gkocik) [reply src/sdkUnitTests/sdkUnitTests.sharpmake.cs]: done

File: src/sdk/sdk.sharpmake.cs (hunk 2 above)
Comment (gkocik) [reply src/sdk/sdk.sharpmake.cs]: done.

**Review by satirev — APPROVED** [review]

**Pull request merged**